annotation/tests/test_tasks_services.py

```diff
@@ -1,30 +1,223 @@
+    ) as mock:
+        yield mock
+
+
+@pytest.fixture
+def mock_agreement_score_response():
+    mock_response = [
+        AgreementScoreServiceResponse(
+            job_id=1, task_id=1, agreement_score=[], annotator_id=uuid.uuid4()
+        ),
+        AgreementScoreServiceResponse(
+            job_id=2, task_id=1, agreement_score=[], annotator_id=uuid.uuid4()
+        ),
+    ]
+    mock_response[0].task_id = 1
+    mock_response[0].agreement_score = [
+        ResponseScore(task_id=2, agreement_score=0.95),
+        ResponseScore(task_id=3, agreement_score=0.8),
+    ]
+    mock_response[1].task_id = 2
+    mock_response[1].agreement_score = [
+        ResponseScore(task_id=1, agreement_score=0.95),
+        ResponseScore(task_id=3, agreement_score=0.85),
+    ]
+    yield mock_response
+
+
+@pytest.fixture
+def create_task():
+    def _create_task(status: TaskStatusEnumSchema):
+        return ManualAnnotationTask(status=status, id=1)
+
+    yield _create_task
+
+
+@pytest.fixture
+def setup_data():
+    task_data = {
+        "objects": [
+            {"id": 1, "name": "Object A", "value": "Some Value", "links": [2]},
+            {"id": 2, "name": "Object B", "value": "Another Value"},
+        ]
+    }
+
+    all_tasks = {
+        1: [
+            ({"name": "Object A", "value": "Some Value"}, "1"),
+            ({"name": "Object C", "value": "Different Value"}, "3"),
+        ],
+        2: [({"name": "Object B", "value": "Another Value"}, "2")],
+    }
+    yield task_data, all_tasks
+
+
+@pytest.fixture
+def mock_session():
+    with patch("annotation.tasks.services.Session", spec=True) as mock_session:
+        yield mock_session()
+
+
+@pytest.fixture
+def mock_validation_revisions():
+    with patch(
+        "annotation.tasks.services.get_file_path_and_bucket",
+        return_value=("s3/path", "bucket"),
+    ) as mock_get_file_path_and_bucket, patch(
+        "annotation.tasks.services.get_annotation_tasks", return_value={}
+    ) as mock_get_annotation_tasks, patch(
+        "annotation.tasks.services.construct_annotated_pages",
+        return_value=([], set()),
+    ) as mock_construct_annotated_pages, patch(
+        "annotation.tasks.services.construct_annotated_doc", return_value=None
+    ) as mock_construct_annotated_doc, patch(
+        "annotation.tasks.services.update_task_status", return_value=None
+    ) as mock_update_task_status, patch(
+        "annotation.tasks.services.Logger.exception", return_value=None
+    ) as mock_logger_exception:
+        yield {
+            "mock_get_file_path_and_bucket": mock_get_file_path_and_bucket,
+            "mock_get_annotation_tasks": mock_get_annotation_tasks,
+            "mock_construct_annotated_pages": mock_construct_annotated_pages,
+            "mock_construct_annotated_doc": mock_construct_annotated_doc,
+            "mock_update_task_status": mock_update_task_status,
+            "mock_logger_exception": mock_logger_exception,
+        }
```

Comment:
it is way too complex, dont use dictionary. split it into separate fixtures

Reply:
I'll do it in the next PR, when I add the function that uses it.

Reply:
Let's do that right now, you add this fixture here